src/Parsers/Access/ASTCreateUserQuery.cpp

```diff
@@ -18,9 +18,17 @@ namespace
                       << quoteString(new_name);
     }
 
-    void formatAuthenticationData(const ASTAuthenticationData & auth_data, const IAST::FormatSettings & settings)
+    void formatAuthenticationData(const std::vector<std::shared_ptr<ASTAuthenticationData>> & auth_data, const IAST::FormatSettings & settings)
     {
-        auth_data.format(settings);
+        auth_data[0]->format(settings);
+
+        auto settings_with_additional_authentication_method = settings;
+        settings_with_additional_authentication_method.additional_authentication_method = true;
```

Comment:
I don't like this approach very much, ideas on how to improve it? One option would be to pollute the `ASTAuthenticationData` and `AuthenticationData` with a field like `bool is_add_identified`. I don't like this approach very much.

Reply:
The easiest approach I can see is to do something like
```cpp
for (size_t i = 1; i < authentication_methods.size(); ++i)
{
    settings.ostr << (settings.hilite ? IAST::hilite_keyword : "") << " ADD" << (settings.hilite ? IAST::hilite_none : "");
    authentication_methods[i]->format(settings);
}
```